### ISSUE TITLE
`Layout/FirstParameterIndentation` can infinite-loop with `Layout/ParameterAlignment: with_fixed_indentation`

### ISSUE THREAD

**Issue by RedZapdos123** [rubocop/rubocop#15639]
## Description:

`Layout/FirstParameterIndentation` can fight with `Layout/ParameterAlignment` when fixed indentation is enabled.
The two cops keep changing the same parameter indentation back and forth, which causes an infinite autocorrect loop.

## Expected behavior:

`Layout/FirstParameterIndentation` should not enter an autocorrect loop with `Layout/ParameterAlignment` when `EnforcedStyle: with_fixed_indentation` is configured.

This file should autocorrect cleanly:
```ruby
def foo(
         a,
         bb
)
end
```

with this configuration:
```yaml
AllCops:
  NewCops: disable

Layout/ParameterAlignment:
  EnforcedStyle: with_fixed_indentation
Layout/FirstParameterIndentation:
  EnforcedStyle: align_parentheses
```

## Actual behavior:

`Layout/FirstParameterIndentation` and `Layout/ParameterAlignment` disagree about the indentation base for this case.

`Layout/FirstParameterIndentation` with `EnforcedStyle: align_parentheses` corrects the first parameter relative to the opening parenthesis, while `Layout/ParameterAlignment` with `EnforcedStyle: with_fixed_indentation` corrects the parameters relative to the method definition line.

As a result, RuboCop enters an autocorrect loop and reports:
```text
Infinite loop detected in example.rb and caused by Layout/FirstParameterIndentation -> Layout/ParameterAlignment

1 error occurred:
Infinite loop detected in example.rb and caused by Layout/FirstParameterIndentation -> Layout/ParameterAlignment
```

## Steps to reproduce the problem:

Create `.rubocop.yml`:
```yaml
AllCops:
  NewCops: disable

Layout/ParameterAlignment:
  EnforcedStyle: with_fixed_indentation
Layout/FirstParameterIndentation:
  EnforcedStyle: align_parentheses
```

Create `example.rb`:
```ruby
def foo(
         a,
         bb
)
end
```

Run:
```console
bundle exec ruby exe/rubocop example.rb --autocorrect --only Layout/ParameterAlignment,Layout/FirstParameterIndentation -f simple
```

Current `master` enters an infinite autocorrect loop for this case.

## RuboCop version:
```console
1.90.0 (using Parser 3.3.12.0, rubocop-ast 1.50.0, analyzing as Ruby 2.7, running on ruby 3.3.11) +Rubydex [x86_64-linux]
  - rubocop-performance 1.27.0
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.2
```
